### ISSUE TITLE
feature request: make a simple protocol / API to allow servers to self report, for indexing

### ISSUE THREAD

**Issue by elendee** [quackduck/devzat#84]
goal:  allow for easy browsing of all devzat active installs, similar to 3rd party "disboard" lists for discord.

the MUD gaming community has something called "global mud protocol"  that allows you to ping servers for current status, if the server admin has implemented it. 

basically this would allow devzat to feel like a single, scaled service even as it potentially scales to lots of individual hosts.

**Comment by quackduck** [comment]
That’s an interesting idea!

**Comment by elendee** [comment]
yea even better.. I was thinking this would be a 3rd party listing, a website etc, but you could build it right into devzat core, as long as there was a registry server online for it to ping

**Comment by l3gacyb3ta** [comment]
Woah, sorta like IRC!

**Comment by elendee** [comment]
different level of abstraction... IRC is a protocol where devzat is an SSH application

**Comment by quackduck** [comment]
What server info should be included in a reply to such a ping?

**Comment by elendee** [comment]
Just self-reported, publicly available info I think.
So I'd think
- created date
- # active users (there is no "registered users" in devzat right... although you could track all-time logins?)
- name
- description
- owner handle (I'd make the owner type this explicitly during config so it's not an email etc by accident)
- sub-channels / topics available

**Comment by quackduck** [comment]
What do you mean by an owner handle? A GitHub/Twitter username?

**Comment by elendee** [comment]
I take handle to mean a "URI safe username", similar to a unique key for an asset in a URI often being called a "slug".

I don't think there's a hard definition.  But I would call "bob" a username, "bob jones" a name, and "bob#1234" a handle

**Comment by quackduck** [comment]
Yeah we can let the owner decide what they'd like to show. I think we can have this work over HTTP as JSON. We'd need to decide a default port for it.

**Comment by quackduck** [comment]
I wonder if we could detect an HTTP client on the SSH port and reply with this message instead of the usual SSH exchange.

**Comment by elendee** [comment]
That does seem ideal yea.  
I think the next best option would be digging into some server-side language that has an SSH package, so you could expose it via that.  That's what I did for MUD servers, using the node telnet package to repackage all the telnet data into a normal node server.   I don't know if SSH has good packages for that though

**Comment by CodeLongAndProsper90** [comment]
You can access argv with an ssh command, correct? Just add an option called '--dump-info' and send the info as json across stdout
